php-classes/Slate/Courses/SectionsRequestHandler.php

```diff
@@ -110,11 +111,52 @@ public static function handleRecordRequest(ActiveRecord $Section, $action = fals
                 ]));
             case 'students':
                 return static::handleStudentsRequest($Section);
+            case false:
+                return static::handleBlogpostRequest($Section, $action);
             default:
                 return parent::handleRecordRequest($Section, $action);
         }
     }
 
+    public static function handleBlogpostRequest(Section $Section, $action)
+    {
+      $className = static::$recordClass;
+
+      $limit = isset($_REQUEST['limit']) && ctype_digit($_REQUEST['limit']) ? (integer)$_REQUEST['limit'] : 10;
+      $offset = isset($_REQUEST['offset']) && ctype_digit($_REQUEST['offset']) ? (integer)$_REQUEST['offset'] : 0;
+      $handle = $_REQUEST['blog_tag'];
+
+      $conditions = [];
+      $latestTeacherPost = false;
+      $tag = null;
+
+      if ($handle) {
+          $tag = Tag::getByHandle($handle);
+
+          if (!$tag) {
+              static::throwNotFoundError('tag not found');
```

Comment:
```suggestion
              return static::throwNotFoundError('tag not found');
```

response calls need to be returned to ensure the function exits and because some response modes return data up the call chain